watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,495 @@
+        ("conc_mol_phase_comp", ("Liq", "Cl_-")): 34.188,
+    }  # Corresponding to C_feed = 2g/L
+    m.fs.feed.properties.calculate_state(
+        init_arg,
+        hold_state=True,
+    )
+    m.fs.EDstack.voltage_applied[0].fix(10)
+    m.fs.recovery_vol_H2O.fix(0.7)
+    condition_base(m)
+
+    # Initialize and solve the model
+    initialize_system(m, solver=solver)
+    solve(m, solver=solver)
+    print("\n***---Fully-defined simulation results, Fixed inlet and voltage---***")
+    display_model_metrics(m)
+
+    # Perform an optimization over cell_length, cell_pair_mum, and voltage_applied
+    ed = m.fs.EDstack
+    ulim = (
+        ed.voltage_x[0, 0].value
+        / ed.current_density_x[0, 0].value
+        * ed.current_dens_lim_x[0, 0].value
+        * 1.5
+    )
+    m.fs.prod.properties[0].conc_mol_phase_comp["Liq", "Na_+"].fix(34.188 * 0.2)
+    m.fs.EDstack.voltage_applied[0].unfix()
+    solve(m, solver=solver, tee=False)
+    m.fs.EDstack.voltage_applied[0].unfix()
+    m.fs.EDstack.voltage_applied[0].setlb(0.1)
+    m.fs.EDstack.voltage_applied[0].setub(ulim)
+    m.fs.EDstack.cell_pair_num.unfix()
+    m.fs.EDstack.cell_pair_num.setlb(1)
+    m.fs.EDstack.cell_pair_num.setub(1000)
+    m.fs.EDstack.cell_length.unfix()
+    m.fs.prod.properties[0].conc_mol_phase_comp["Liq", "Na_+"].fix(
+        1.7094
+    )  # Corresponding to C_product = 100 ppb
+    m.fs.objective = Objective(expr=m.fs.costing.LCOW)
+    solve(m, solver=solver, tee=True)
+    m.fs.EDstack.cell_pair_num.fix(round(m.fs.EDstack.cell_pair_num.value))
+    solve(m, solver=solver, tee=True)
+    print("\n***---Optimization results, Product conc = 100 ppb---***")
+    display_model_metrics(m)
+
+    return m
+
+
+def build():
+    # ---building model---
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    ion_dict = {
+        "solute_list": ["Na_+", "Cl_-"],
+        "mw_data": {"H2O": 18e-3, "Na_+": 23e-3, "Cl_-": 35.5e-3},
+        "elec_mobility_data": {("Liq", "Na_+"): 5.19e-8, ("Liq", "Cl_-"): 7.92e-8},
+        "diffusivity_data": {("Liq", "Na_+"): 1.33e-9, ("Liq", "Cl_-"): 2.03e-9},
+        "charge": {"Na_+": 1, "Cl_-": -1},
+    }
+    m.fs.properties = MCASParameterBlock(**ion_dict)
+    m.fs.costing = WaterTAPCosting()
+    m.fs.feed = Feed(property_package=m.fs.properties)
+    m.fs.sepa0 = Separator(
+        property_package=m.fs.properties,
+        outlet_list=["to_dil_in", "to_conc_in0"],
+    )
+    m.fs.mix0 = Mixer(
+        property_package=m.fs.properties,
+        energy_mixing_type=MixingType.none,
+        momentum_mixing_type=MomentumMixingType.none,
+        inlet_list=["from_feed", "from_conc_out"],
+    )
+
+    m.fs.pump0 = Pump(property_package=m.fs.properties)
+    m.fs.pump0.del_component("ratioP")
+    m.fs.pump0.del_component("ratioP_calculation")
+    m.fs.pump1 = Pump(property_package=m.fs.properties)
+    m.fs.pump1.del_component("ratioP")
+    m.fs.pump1.del_component("ratioP_calculation")
```

Comment:
This is a bit weird -- should we change the `Pump` unit?

Reply:
I del the var and its corresponding constraint because I don't really need them for my model.  Keeping them will not cause an error but create unnecessary computation. I would keep this as is now.

Reply:
Recommendation accepted.